rfcs/InputUnion.md

```diff
@@ -555,6 +555,7 @@ input DogInput {
   * ✅
 * [B. Input polymorphism matches output polymorphism][criteria-b]
   * ✅ Data structures can mirror eachother.
+  * ⚠️ `__typename` can not match discriminator since Input & Output types are distinct (ex: `Cat` vs `CatInput`).
```

Comment:
I'm not sure this is true - this solution is suggesting a pre-defined "literal" value as the discriminator, not the GraphQL type name.

Perhaps the confusion is that the first "variation" here is actually using the type name, I should probably just remove that in favor of the string / enum literal variations.

I think that's the main point of the solution, to enable the same data to work on query &. mutation sides...

Reply:
Ah, yeah I definitely saw the type names in the example and made that assumption, it would be good to clarify.